### PR TITLE
DOC: update the DataFrame.plot.line docstring

### PR DESCRIPTION
**Pull request by ifosch**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```❯ python scripts/validate_docstrings.py pandas.DataFrame.plot.line

################################################################################
#################### Docstring (pandas.DataFrame.plot.line) ####################
################################################################################

Plot DataFrame columns as lines.

This function is useful to plot lines using DataFrame's values
as coordinates.

Parameters
----------
x : int or str, optional
    Columns to use for the horizontal axis.
    Either the location or the label of the columns to be used.
    By default, it will use the `DataFrame` indices.
y : int, str, or list of them, optional
    The values to be plotted.
    Either the location or the label of the columns to be used.
    By default, it will use the remaining DataFrame numeric columns.
kwds : optional
    Keyword arguments to pass on to :py:meth:pandas.DataFrame.plot.

Returns
-------
axes : :class:matplotlib.AxesSubplot or :class:numpy.ndarray of
       them.

See Also
--------
matplotlib.pyplot.plot : Plot y versus x as lines and/or markers.

Examples
--------

.. plot::
    :context: close-figs

    The following example shows the populations for some animals
    over the years.

    >>> df = pd.DataFrame({
    ...    'pig': [20, 18, 489, 675, 1776],
    ...    'horse': [4, 25, 281, 600, 1900]
    ...    }, index=[1990, 1997, 2003, 2009, 2014])
    >>> lines = df.plot.line()

.. plot::
    :context: close-figs

    The following example shows the relationship between both
    populations.

    >>> lines = df.plot.line(x='pig', y='horse')

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.plot.line" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
 
     def line(self, x=None, y=None, **kwds):
         """
-        Line plot
+        Plot `DataFrame` columns as lines.
+
+        This function is useful to plot lines using `DataFrame`'s values
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: I think sphinx complains if you do you have something right after a \`. Just do DataFrame without the backticks.

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+        as coordinates.
+        It allows to specify which columns to use for the axes.
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Can remove this line since we discuess that in the parameters section.

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : int, str, list of int or list of str, optional
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: I think x has to be an int or str. Lists are not allowed for x, just y.

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+            It defines the X axis coordinates.
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Remove "it". Just have "Column to use for the horizontal axis." and then the rest of your description.

File: pandas/plotting/_core.py [hunk 5]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+            Either the location or the label of the columns to be used.
+            By default, it will use the `DataFrame` indices.
+        y : int, str, list of int or list of str, optional
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: I thin we write this as `int, str, or list of them, optional`.

File: pandas/plotting/_core.py [hunk 6]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+            It defines the values to be plotted.
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: "The values to be plotted."

File: pandas/plotting/_core.py [hunk 7]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+            Either the location or the label of the columns to be used.
+            By default, it will use the remaining `DataFrame` columns.
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: No backticks. Specify that it's the remaining *numeric* columns.

File: pandas/plotting/_core.py [hunk 8]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+        kwds : optional
             Keyword arguments to pass on to :py:meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : matplotlib.AxesSubplot or np.array of them
+        axes : :class:`matplotlib.AxesSubplot` or :meth:`np.array` of
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: :meth:`np.array` -> ndarray. I don't care either way about the sphinx markup so either

    :class:numpy.ndarray

or

    numpy.ndarray

File: pandas/plotting/_core.py [hunk 9]
```diff
@@ -2704,18 +2704,56 @@ def __call__(self, x=None, y=None, kind='line', ax=None,
...
+            ...    }, index=[1990, 1997, 2003, 2009, 2014])
+            >>> lines = df.plot.line()
+
+        .. plot::
+            :context: close-figs
+
+            The following example shows the relationship between both
+            populations.
+
+            >>> lines = df.plot.line('pig', 'horse')
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Specify `x='pig', y='horse'`

File: pandas/plotting/_core.py (hunk 3 above)
Comment (ifosch) [reply pandas/plotting/_core.py]: We tested on Jupyter and these seem to be allowed. However, the result makes not much sense. I'm changing this, anyway.

**Comment by ifosch** [comment]
Fixes are up now

**Review by TomAugspurger — APPROVED** [review]
Pushed a couple small updates.

Will wait to see if anyone else has thoughts, otherwise ready to merge.

**Comment by ifosch** [comment]
Thanks @TomAugspurger !

**Comment by TomAugspurger** [comment]
Thank you!

**Pull request merged**